software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessSshDriver.java

```diff
@@ -285,45 +296,54 @@ public void copyInstallResources() {
         }
     }
 
+    private void executeSuccessfully(ConfigKey<String> configKey, String label) {
+        log.debug("Executing {} on entity {}", label, entity.getDisplayName());
+        int result = execute(ImmutableList.of(getEntity().getConfig(configKey)), label);
+        if (0 != result) {
+            log.debug("Executing {} failed with return code {}", label, result);
+            throw new IllegalStateException("commands for " + configKey.getName() + " failed with return code " + result);
+        }
+    }
+
     @Override
     public void runPreInstallCommand() {
         if(Strings.isNonBlank(getEntity().getConfig(BrooklynConfigKeys.PRE_INSTALL_COMMAND))) {
```

Comment:
Could move all of these checks to the `executeSuccessfully` method, but happy to merge as is.